app/Platform/Actions/BuildGameShowPagePropsAction.php

```diff
@@ -31,15 +33,43 @@ public function __construct(
         protected BuildGameAchievementDistributionAction $buildGameAchievementDistributionAction,
         protected LoadGameTopAchieversAction $loadGameTopAchieversAction,
         protected BuildSeriesHubDataAction $buildSeriesHubDataAction,
+        protected ResolveBackingGameForAchievementSetAction $resolveBackingGameForAchievementSetAction,
     ) {
     }
 
-    public function execute(Game $game, ?User $user): GameShowPagePropsData
+    public function execute(Game $game, ?User $user, ?int $targetAchievementSetId = null): GameShowPagePropsData
     {
-        [$numMasters, $topAchievers] = $this->loadGameTopAchieversAction->execute($game);
+        // The backing game is the legacy game that backs the target achievement set.
+        // For core sets, this will be $game->id. For subsets, it'll be a different ID.
+        $backingGameId = null;
+
+        if ($targetAchievementSetId !== null) {
+            $backingGameId = $this->resolveBackingGameForAchievementSetAction->execute($targetAchievementSetId);
+        }
+
+        // If we have a backing game ID different from the current game, load it.
+        // Otherwise, use the current game as the backing game.
+        $backingGame = null;
+        if ($backingGameId && $backingGameId !== $game->id) {
+            $backingGame = Game::find($backingGameId);
+
+            // Load the visible comments relationship for the backing game.
+            $backingGame->load(['visibleComments' => function ($query) {
+                $query->latest('Submitted')
+                    ->limit(20)
+                    ->with(['user' => function ($userQuery) {
+                        $userQuery->withTrashed();
+                    }]);
+            }]);
```

Comment:
Adding a comment to the subset page posts it on the main game page.

Reply:
Fixed in latest.